packages/cli/src/environments/sourceControl/sourceControlImport.service.ee.ts

```diff
@@ -351,10 +351,21 @@ export class SourceControlImportService {
 				this.logger.debug(`Updating credential id ${newCredentialObject.id as string}`);
 				await Container.get(CredentialsRepository).upsert(newCredentialObject, ['id']);
 
-				if (!sharedOwner) {
+				const isOwnedLocally = existingSharedCredentials.some(
+					(c) => c.credentialsId === credential.id,
+				);
+
+				if (!isOwnedLocally) {
+					const remoteOwnerId = await Container.get(UserRepository)
+						.findOne({
+							where: { email: credential.ownedBy },
+							select: { id: true },
+						})
+						.then((user) => user?.id);
```

Comment:
```suggestion
					const remoteOwnerId = credential.ownedBy ? await Container.get(UserRepository)
						.findOne({
							where: { email: credential.ownedBy },
							select: { id: true },
						})
						.then((user) => user?.id) : undefined;
```

In case email is empty

Reply:
Addressed at [2746b6a](https://github.com/n8n-io/n8n/pull/8955/commits/2746b6a273d7248a94d08a226da843ae90aa1bd4)